Assets/Mirror/Runtime/NetworkReaderPool.cs

```diff
@@ -1,41 +1,63 @@
-﻿using System;
+using System;
 using System.Collections.Generic;
 
 namespace Mirror
 {
-    public static class NetworkReaderPool
+    public class NetworkReaderPool : NetworkReader, IDisposable
     {
-        static readonly Stack<NetworkReader> pool = new Stack<NetworkReader>();
+        private NetworkReaderPool(byte[] bytes) : base(bytes) { }
 
-        public static NetworkReader GetReader(byte[] bytes)
+        private NetworkReaderPool(ArraySegment<byte> segment) : base(segment) { }
+
+        static readonly Stack<NetworkReaderPool> pool = new Stack<NetworkReaderPool>();
+
+        public static NetworkReaderPool GetReader(byte[] bytes)
```

Comment:
why did you decide to have the pool and the pooled reader in one class?

Reply:
I followed the model of NetworkWriterPool.